File: p.112/Python/solution.py
def is_increasing(num):
    s = str(num)
    return all(s[i] <= s[i+1] for i in range(len(s) - 1))

def is_decreasing(num):
    s = str(num)
    return all(s[i] >= s[i+1] for i in range(len(s) - 1))

def is_bouncy(num):
    return not (is_decreasing(num) or is_increasing(num))

count_bouncy = 0
n = 99
while True:
    n += 1
    if is_bouncy(n):
        count_bouncy += 1
    if (count_bouncy * 100) == 99 * n:
        print(n)
        break

File: p.112/Rust/solution.rs
fn is_increasing(num: u64) -> bool {
    let s = num.to_string();
    s.as_bytes().windows(2).all(|pair| pair[0] <= pair[1])
}

fn is_decreasing(num: u64) -> bool {
    let s = num.to_string();
    s.as_bytes().windows(2).all(|pair| pair[0] >= pair[1])
}

fn is_bouncy(num: u64) -> bool {
    !(is_increasing(num) || is_decreasing(num))
}

fn main() {
    let mut count_bouncy = 0u64;
    let mut n = 99u64;

    loop  {
        n += 1;
        if is_bouncy(n) {
            count_bouncy += 1;
        }

        if count_bouncy * 100 == 99 * n {
            println!("{}", n);
            break;
        }
    }
}
